hraven-core/src/main/java/com/twitter/hraven/rest/client/HRavenRestClient.java

```diff
@@ -152,7 +247,93 @@ public String getCluster(String hostname) throws IOException {
     return retrieveFlowsFromURL(urlString);
   }
 
-  private String buildConfigParam(String paramName, String[] paramArgs) throws IOException {
+  public List<Flow> fetchFlowsWithConfigPatterns(String cluster,
+                                                 String username,
+                                                 String batchDesc,
+                                                 String signature,
+                                                 int limit,
+                                                 List<String> flowResponseFilters,
+                                                 List<String> jobResponseFilters,
+                                                 List<String> configPatterns)
+                                                     throws IOException {
+    LOG.info(String.format(
+        "Fetching last %d matching jobs for cluster=%s, user.name=%s, "
+            + "batch.desc=%s, pig.logical.plan.signature=%s", limit, cluster,
+        username, batchDesc, signature));
+
+    StringBuilder urlStringBuilder = buildFlowURL(cluster, username, batchDesc,
+        signature, limit, flowResponseFilters, jobResponseFilters);
+
+    if ((configPatterns != null) && (configPatterns.size() > 0)) {
+      urlStringBuilder.append(AND);
+      urlStringBuilder.append(buildParam("includeConfRegex",
+          (String[]) configPatterns.toArray()));
+    }
+    return retrieveFlowsFromURL(urlStringBuilder.toString());
+  }
+
+  /**
+   * builds up a StringBuilder with the parameters for the FLOW API
+   * @param cluster
+   * @param username
+   * @param batchDesc
+   * @param signature
+   * @param limit
+   * @param flowResponseFilters
+   * @param jobResponseFilters
+   * @return
+   * @throws IOException
+   */
+  private StringBuilder buildFlowURL(String cluster,
```

Comment:
Nit: it can simply return the final String (instead of StringBuilder), right? Might be better to return an immutable string if you're not going to modify things further.